torch/_higher_order_ops/cond.py

```diff
@@ -107,10 +111,19 @@ def false_fn(x: torch.Tensor):
         - The **output** of branches must be a **single Tensor**. Pytree of tensors will be supported in the future.
 
     """
-
     if torch.compiler.is_dynamo_compiling():
         return cond_op(pred, true_fn, false_fn, operands)
 
+    if isinstance(pred, (bool, int, float)):
+        log.warning(
+            "Pred is a Python constant. When used with torch.cond, it executes only one of the branches."
+            " If you want torch.cond to perserve two branches, please make the predicate a boolean tensor."
```

Comment:
can cond take symbool?

Reply:
yeah, add a test_export_cond_symbool_pred to reflect this. cond won't specialize on symbool.